### PR TITLE
Address garbage collection for lasers and space rocks

### PR DESCRIPTION
**Pull request by ToddiusNoobius**
Lasers and space rocks are garbage collected to improve game performance

### CODE REVIEW

File: seagulls-space-shooter-demo/src/seagulls/space_shooter/_laser.py [hunk 1]
```diff
@@ -40,7 +42,6 @@ def render(self, surface: Surface) -> None:
 
         surface.blit(laser_sprite, self._position)
 
-    @lru_cache()
     def _get_cached_laser(self) -> Surface:
```
Comment (Zeelot) [inline seagulls-space-shooter-demo/src/seagulls/space_shooter/_laser.py:45]: should we rename this method now that it's not cached? maybe clarify a bit more and call it `_get_sprite()`?

File: seagulls-space-shooter-demo/src/seagulls/space_shooter/_ship_powered_up_rule.py [hunk 2]
```diff
@@ -19,4 +19,4 @@ def check(self) -> None:
             self._state_client.update_state(True)
 
     def _score_threshold_check(self) -> bool:
-        return self._score_tracker.get_score() >= 75
+        return self._score_tracker.get_score() >= 20
```
Comment (Zeelot) [inline seagulls-space-shooter-demo/src/seagulls/space_shooter/_ship_powered_up_rule.py:22]: do we want to set this back to 75 before merging?

File: seagulls-space-shooter-demo/src/seagulls/space_shooter/_space_rock.py [hunk 3]
```diff
@@ -45,7 +44,6 @@ def render(self, surface: Surface) -> None:
         else:
             surface.blit(self._get_cached_rock_large(), self._position)
 
-    @lru_cache()
     def _get_cached_rock_small(self) -> Surface:
```
Comment (Zeelot) [inline seagulls-space-shooter-demo/src/seagulls/space_shooter/_space_rock.py:47]: same here :) just `_get_small_sprite()`? with similar changes to methods below?

**Review by Zeelot — APPROVED** [review]

File: seagulls-space-shooter-demo/src/seagulls/space_shooter/_space_rock.py (hunk 3 above)
Comment (ToddiusNoobius) [reply seagulls-space-shooter-demo/src/seagulls/space_shooter/_space_rock.py]: Sure thing - updated

File: seagulls-space-shooter-demo/src/seagulls/space_shooter/_ship_powered_up_rule.py (hunk 2 above)
Comment (ToddiusNoobius) [reply seagulls-space-shooter-demo/src/seagulls/space_shooter/_ship_powered_up_rule.py]: Yep, updated

File: seagulls-space-shooter-demo/src/seagulls/space_shooter/_laser.py (hunk 1 above)
Comment (ToddiusNoobius) [reply seagulls-space-shooter-demo/src/seagulls/space_shooter/_laser.py]: Yep, updated

**Pull request merged**